packages/n4s/src/runtime/enforceEager.ts

```diff
@@ -42,28 +51,42 @@ export default function enforceEager(value: RuleValue): IRules {
   // This function is used to wrap a rule with the base enforce behavior
   // It takes the target object, the rule function, and the rule name
   // It then returns the rule, in a manner that can be used by enforce
-  function genRuleCall(target: IRules, rule: RuleBase, ruleName: string) {
+  function genRuleCall(
+    target: enforceEagerReturn,
+    rule: RuleBase,
+    ruleName: string
+  ) {
     return function ruleCall(...args: Args) {
       // Order of operation:
       // 1. Create a context with the value being enforced
       // 2. Call the rule within the context, and pass over the arguments passed to it
       // 3. Transform the result to the correct output format
-      const transformedResult = ctx.run({ value }, () =>
-        transformResult(rule(value, ...args), ruleName, value, ...args)
-      );
+      const transformedResult = ctx.run({ value }, () => {
+        return transformResult(rule(value, ...args), ruleName, value, ...args);
+      });
+
+      function renderMessage() {
+        const shouldUseCustomMessage = !isNullish(customMessage);
+        if (shouldUseCustomMessage) return customMessage;
+        if (isNullish(transformedResult.message)) {
+          return `enforce/${ruleName} failed with ${JSON.stringify(value)}`;
+        } else {
+          return StringObject(transformedResult.message);
+        }
```

Comment:
The else is unneeded here
```suggestion
        }

        return StringObject(transformedResult.message);
```